### DISCUSSION TITLE
create embedded document with sql-insert

### DISCUSSION THREAD

**Discussion by gfinger** [ArcadeData/arcadedb#969] [Q&A] [answered]
I created a database schema with:

``` sql
create vertex type Address if not exists;
create property Address.city if not exists string;
create vertex type Customer if not exists;
create property Customer.name if not exists string;
create property Customer.address if not exists embedded;
``` 

but when I send 

```
insert into Customer content {
    "name": "Happy Garden",
    "address": {
        "street": "Flower Road",
        "zip": "12345",
        "city": "San Francisco",
        "phone": "012367"
    }
}
```

I get an error saying:

```
Cannot convert type 'class java.util.HashMap' to 'EMBEDDED' found in property 'address'
```

If I leave out the creation of the embedded property address in the schema, the insert works well. How should an embedded document be defined in the schema, if at all? How is the content in the insert request supposed to look like?

**Comment by gramian** [comment, marked as answer]
Hi,
I think you need to define and add a `@type` key-value pair  to the address document, so:
```
create document type AddressDoc;
```
and:
```
insert into Customer content {
    "name": "Happy Garden",
    "address": {
        "@type": "AddressDoc",
        "street": "Flower Road",
        "zip": "12345",
        "city": "San Francisco",
        "phone": "012367"
    }
}
```
If you don't add the `@type` to your inserted address content, it is a `Map`.

**Comment by gfinger** [comment]
Yes, super, that's it! Thank you for the help!
Additionally I also needed to change the definition of Address in the Schema from Vertex Type to Document Type.
